app/waf/src/controllers/application.controller.ts

```diff
@@ -250,6 +259,37 @@ export class ApplicationController {
       }
     }
 
+    let rules = [];
+    let wafs = [];
+    if (service) {
+      params.service = service;
+      let endpointpolicyId = service.endpointpolicy;
+      if (endpointpolicyId !== undefined ) {
+        let endpointpolicy = await this.endpointpolicyRepository.findById(
+          endpointpolicyId,
+        );
+        if (endpointpolicy) {
```

Comment:
I prefer to use less if to replace `if (endpointpolicy) {`, as this: 
```
    if (service.endpointpolicy) {
      params.endpointpolicy = await this.endpointpolicyRepository.findById(
        service.endpointpolicy,
      );

      params.rules = await this.ruleRepository.find({
        where: {
          id: {
            inq: (<Endpointpolicy>params.endpointpolicy).rules,
          },
        },
      });

      params.wafs = await this.wafpolicyRepository.find({
        where: {
          id: {
            inq: (<Array<Rule>>params.rules).map(v => {
              return v.wafpolicy;
            }),
          },
        },
      });
    }
```

Reasons for this are:
1. easier to make them covered. As you know, if else is not easier to cover. 
2. use `find` is more powerful, at least no exceptions. 
3. codes look clear.

Reply:
I will try with this.